crates/bin/prove_block/src/reexecute.rs

```diff
@@ -40,10 +41,25 @@ fn get_tx_hash(tx: &Transaction) -> TransactionHash {
 pub fn reexecute_transactions_with_blockifier<S: StateReader>(
     state: &mut CachedState<S>,
     block_context: &BlockContext,
+    previous_block_hash: Felt252,
```

Comment:
Dont call this `previous_block_hash`. 
Is misleading as we are calling previous to current - 1 block

`old_block_hash` its ok. But  `buffer_block_hash` would be ideal